sender.go

```
@@ -52,9 +62,34 @@ var gcmSendEndpoint = GcmSendEndpoint
 //	}
 type Sender struct {
 	ApiKey string
+	URL    string
 	Http   *http.Client
 }
 
+// NewClient returns a new sender with the given URL and apiKey.
+// If one of input is empty or URL is malformed, returns error.
+// It sets http.DefaultHTTP client for http connection to server.
+// If you need our own configuration overwrite it.
+func NewClient(urlString, apiKey string) (*Sender, error) {
+	if len(urlString) == 0 {
+		return nil, fmt.Errorf("missing GCM/FCM endpoint url")
+	}
+
+	if len(apiKey) == 0 {
+		return nil, fmt.Errorf("missing API Key")
+	}
+
+	if _, err := url.Parse(urlString); err != nil {
+		return nil, fmt.Errorf("failed to parse URL %q: %s", urlString, err)
+	}
+
+	return &Sender{
+		URL:    urlString,
+		ApiKey: apiKey,
+		Http:   http.DefaultClient,
```

IMO, using http.DefaultClient is bad approach. Because http.DefaultClient.Timeout is zero by default. It means no timeout and it should not set in production server.

---

- It's not related to this PR (by default, this package use default client)
- (even if it's related) It's not what package author should handle. Since `Http` filed is exported, user can replace it with their own HTTP client with their favorite timeout or what ever. See other client package, most of them use default client and just allow to modify it on user side.

---

Why don't you give customized **http.Client** to **NewClient()**?

---

I've never seen such API client that passes `http.Client` to `NewClient()` .... I read some of API client implementation and found current `NewClient` interface was kind common for me ( and I hope for some other gophers). 

If you really need to provide API for changing it, I would add `SetHTTPClient()` function instead of via `NewClient()`.

---

> If you really need to provide API for changing it, I would add SetHTTPClient() function instead of via NewClient().

That sounds good. It's a deal. And please change the behavior below to return error instead of setting default http.Client.

https://github.com/mercari/gcm/blob/feature/fcm/sender.go#L238